libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx

```diff
@@ -49,22 +49,32 @@ export const RecommendationPanelContext = (props: RecommendationPanelContextProp
 
   const [selectionState, setSelectionState] = useState<SelectionState>(SELECTION_STATES.SEARCH);
 
-  const selectedOperationId = useSelectedSearchOperationId();
   const { data: allOperations, isLoading: isLoadingOperations } = useAllOperations();
-  const selectedOperation = allOperations.find((o) => o.id === selectedOperationId);
+  const [selectedOperation, setSelectedOperation] = useState<DiscoveryOperation<DiscoveryResultTypes> | undefined>(undefined);
 
   const selectedOperationGroupId = useSelectedSearchOperationGroupId();
   const { data: allConnectors } = useAllConnectors();
   const selectedConnector = allConnectors?.find((c) => c.id === selectedOperationGroupId);
 
+  // effect to set the current list of operations by group
   useEffect(() => {
-    if (!allOperations || !selectedOperationGroupId) return;
-    const filteredOps = allOperations.filter((operation) => {
-      const apiId = operation.properties.api.id;
-      return areApiIdsEqual(apiId, selectedOperationGroupId);
+    if (!selectedOperationGroupId) return;
+
+    const searchOperation = SearchService().getOperationsByConnector?.bind(SearchService());
+
+    const searchResultPromise = searchOperation
+      ? searchOperation(selectedOperationGroupId, filters['actionType']?.toLowerCase())
+      : Promise.resolve(
+          (allOperations ?? []).filter((operation) => {
+            const apiId = operation.properties.api.id;
+            return areApiIdsEqual(apiId, selectedOperationGroupId);
+          })
+        );
+
+    searchResultPromise.then((filteredOps) => {
+      setAllOperationsForGroup(filteredOps);
```

Comment:
should we consider some kind of loading state now that this is async, maybe in long term fix

Reply:
Thanks for the suggestion, done.